docs/dev/cli-architecture.md

```diff
@@ -0,0 +1,28 @@
+# Definition CLI Architecture
+
+## Basics
+- parse and call the commands
+- call the method to transform and/ or deploy a given topology to a custom platform (start / stop / transform platform)
+- show current status of transformation
+- print detailed logs if wanted
+- get status from different components
+- show supported platforms
+- create threads to provide asynchronous operations
+
+## supported commands
+| command | description | option |
+|-----------------------|-----------------------|-------------------------------|
+| start transformation | starts the transformation | -startt or -start transformation |
+| stop transformation | stops a transformation | -stopt or -stop transformation |
+| start deployment | starts the deployment | -startd or -start deployment |
+| stop deployment | stops the deployment | -stopd or -stop deployment |
+| status | prints some information about a current transformation or component | -status |
+| verbose | show logs while transformation | -v or -verbose |
+| list | show all available supported platforms | -l or -list |
```

Comment:
Long commands usally are written like this `--long-command` not `-long-command`.